src/arduino/arduino.ts

```diff
@@ -140,8 +140,8 @@ export class ArduinoApp {
             }
         }
 
-        const appPath = path.join(ArduinoWorkspace.rootPath, dc.sketch);
-        const args = ["--upload", "--board", boardDescriptor];
+        const appPath = this.isArduinoCli() ? ArduinoWorkspace.rootPath : path.join(ArduinoWorkspace.rootPath, dc.sketch);

```

Comment:
Not including the sketch path here breaks breaks compilation of sketches in subfolders of the workspace.

Reply:
The path of the sketch can be passed as a parameter but as you said the containing folder needs have the same name as the .ino file. The following two commands yield the same result
```
arduino-cli compile Disco/Disco.ino -b arduino:avr:uno
arduino-cli compile Disco -b arduino:avr:uno
cd Disco; arduino-cli compile -b arduino:avr:uno
```